realm/realm-library/src/objectServer/java/io/realm/mongodb/User.java

```diff
@@ -502,8 +502,8 @@ public boolean equals(@Nullable Object o) {
 
         User user = (User) o;
 
-        if (!osUser.equals(user.osUser)) return false;
-        return app.equals(user.app);
+        if (!osUser.getIdentity().equals(user.osUser.getIdentity())) return false;
+        return app.getConfiguration().getAppId().equals(user.app.getConfiguration().getAppId());
```

Comment:
Does the `app_id` hash include the user identity also? if yes we can simplify the condition to check only on it. If not we need to maintain this and we risk having a collision in https://github.com/realm/realm-java/pull/7095/files#diff-79b05914e49eabc2a48d017ab1b7b38dR119  since two users could have the same app_id?

Reply:
We check the user identity in line 505, so in combination with 506 we should check for the pair `<appId, userId>` which should uniquely identify the user?